bootstrap/config/config_test.go

```diff
@@ -92,3 +108,237 @@ func TestGetSecretPathsChanged(t *testing.T) {
+					assert.NotEmpty(t, serviceConfigMock.Writable.StoreAndForward.RetryInterval)
+					assert.NotZero(t, serviceConfigMock.Writable.StoreAndForward.MaxRetryCount)
+				case config.ServiceTypeDevice:
+					assert.False(t, serviceConfigMock.Writable.Telemetry.Metrics["EventsSent"])
+					assert.True(t, serviceConfigMock.Writable.Telemetry.Metrics["ReadingsSent"])
+				}
+				return
+			}
+			assert.Equal(t, err, tc.expectedErr)
+		})
+	}
+}
+
+func TestMergeConfigs(t *testing.T) {
+	// create the service config
+	serviceConfig := ConfigurationMockStruct{
+		Writable: WritableInfo{
+			LogLevel: "INFO",
+		},
+		Registry: config.RegistryInfo{
+			Host: "localhost",
+			Port: 8500,
+			Type: "consul",
+		},
+	}
+	require.NotEmpty(t, serviceConfig.Writable.LogLevel)
+	require.NotEmpty(t, serviceConfig.Registry.Host)
+	require.NotZero(t, serviceConfig.Registry.Port)
+	require.NotEmpty(t, serviceConfig.Registry.Type)
+
+	// create the app config
+	appConfig := ConfigurationMockStruct{
+		Writable: WritableInfo{
+			StoreAndForward: StoreAndForwardInfo{
+				Enabled:       false,
+				RetryInterval: "5m",
+				MaxRetryCount: 10,
+			},
+		},
+		Trigger: TriggerInfo{
+			Type: "edgex-messagebus",
+		},
+	}
+	require.False(t, appConfig.Writable.StoreAndForward.Enabled)
+	require.NotEmpty(t, appConfig.Writable.StoreAndForward.RetryInterval)
+	require.NotZero(t, appConfig.Writable.StoreAndForward.MaxRetryCount)
+	require.NotEmpty(t, appConfig.Trigger.Type)
+
+	// merge the configs
+	err := mergeConfigs(&serviceConfig, &appConfig)
+	require.NoError(t, err)
+
+	// verify values
+	assert.False(t, serviceConfig.Writable.StoreAndForward.Enabled)
+	assert.NotEmpty(t, serviceConfig.Writable.StoreAndForward.RetryInterval)
+	assert.NotZero(t, serviceConfig.Writable.StoreAndForward.MaxRetryCount)
+	assert.NotEmpty(t, serviceConfig.Trigger.Type)
+}
+
+func TestMergeMaps(t *testing.T) {
+	destMap := map[string]any{
+		"Writable": WritableInfo{
+			StoreAndForward: StoreAndForwardInfo{
+				Enabled:       false,
+				RetryInterval: "5m",
+				MaxRetryCount: 10,
+			},
+		},
+		"Registry": config.RegistryInfo{
+			Host: "localhost",
+			Port: 8500,
+			Type: "consul",
+		},
+		"Service":             config.ServiceInfo{},
+		"HttpServer":          HttpConfig{},
+		"MessageBus":          config.MessageBusInfo{},
+		"Trigger":             TriggerInfo{},
+		"ApplicationSettings": nil,
+		"Clients":             nil,
+	}
+	srcMap := map[string]any{
+		"Writable": WritableInfo{
+			StoreAndForward: StoreAndForwardInfo{
+				Enabled:       false,
+				RetryInterval: "5m",
+				MaxRetryCount: 10,
+			},
+		},
+		"Trigger": TriggerInfo{
+			Type: "edgex-messagebus",
+		},
+	}
+	mergeMaps(destMap, srcMap)
+
+	for key, value := range destMap {
+		if key == "StoreAndForwardInfo" || key == "Trigger" {
+			assert.NotEmpty(t, value)
+		}
+	}
+}
+
+func TestRemoveZeroValues(t *testing.T) {
+	config := struct {
+		MaxEventSize int
+	}{
+		MaxEventSize: 4567,
+	}
```

Comment:
This test is not complete. Need to add additional fields here that would be ZeroValues.

Reply:
I tried that - but in debug mode, as I was watching, the Zero Values get removed automatically? or are the default?

Reply:
I am suggestion this:
```
	config := struct {
		MaxEventSize int

	}{
		MaxEventSize: 4567,
	}
```

Reply:
ok. I fixed this one - by using the configMockStruct